code/renderers/react/src/docs/jsxDecorator.tsx

```diff
@@ -74,21 +72,16 @@ export const renderJsx = (code: React.ReactElement, options: JSXOptions) => {
     }
   }
 
-  const displayNameDefaults =
-    typeof options.displayName === 'string'
-      ? { showFunctions: true, displayName: () => options.displayName }
-      : {
-          // To get exotic component names resolving properly
-          displayName: (el: any): string =>
-            el.type.displayName ||
-            (el.type === Symbol.for('react.profiler') ? 'Profiler' : null) ||
-            getDocgenSection(el.type, 'displayName') ||
-            (el.type.name !== '_default' ? el.type.name : null) ||
-            (typeof el.type === 'function' ? 'No Display Name' : null) ||
-            (isForwardRef(el.type) ? el.type.render.name : null) ||
-            (isMemo(el.type) ? el.type.type.name : null) ||
-            el.type,
-        };
+  let displayNameDefaults;
```

Comment:
**Major Change**: `displayNameDefaults.displayName` should be `undefined` if `getDocgenSection` returns `null`:

https://github.com/storybookjs/storybook/blob/9a07fcba8a581558d0fa8262989ef83b70a507a0/code/lib/docs-tools/src/argTypes/docgen/utils/docgenInfo.ts#L14-L16

and then the logic will be handled by `react-element-to-jsx-string`:

https://github.com/algolia/react-element-to-jsx-string/blob/984f26908c17cea134fa53d17ac51f179e5195b2/src/parser/parseReactElement.js#L47-L77

We don't need to rewrite redundant code here.